Library/ViewModels/ProjectPamphletViewModel.swift

```diff
@@ -80,13 +86,36 @@ public final class ProjectPamphletViewModel: ProjectPamphletViewModelType, Proje
     let freshProjectAndRefTag = freshProjectAndRefTagEvent.values()
 
     let ctaButtonTapped = freshProjectAndRefTag
-      .takeWhen(self.backThisProjectTappedProperty.signal)
-      .map { project, refTag in
-        (project, refTag)
+      .takePairWhen(self.pledgeCTAButtonTappedProperty.signal)
+      .map(unpack)
+
+    let goToManagePledge = ctaButtonTapped
+      .filter(canShowManageViewPledgeScreen(_:_:state:))
```

Comment:
Looks like `canShowManageViewPledgeScreen` does not need `refTag`? Could omit and have more readable call site `canShowManageViewPledgeScreen(_:state:)`

Reply:
The problem with that is `ctaButtonTapped` emits (project, refTag, state). If we omit the refTag, the compiler will complain because the types emitted don't match the function arguments, thus the point free syntax will no longer be possible.

Reply:
I see!

Thanks for clarifying this...I don't know whether a stylistic thing like being able to call something with `point-free` expression would overweight the code debt of passing argument that's not necessary.

In this case, I'd consider whether non-point-free expression would be more suitable (less code + less debt).

```swift
.filter { (project, _, state) in canShowManageViewPledgeScreen(project, state:state) }
```

or

```swift
.filter { canShowManageViewPledgeScreen($0.0, state:$0.2) }
```